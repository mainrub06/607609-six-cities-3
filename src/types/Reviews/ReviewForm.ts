import { ChangeEvent } from "react";

interface IEvent {
  event: ChangeEvent<HTMLInputElement>;
}

export interface IReviewForm {
  activeHotelId: string;
  handleSubmitFeedback: Function;
  reviewsResponse: number;
  isButtonActive: boolean;
  resetForm: Function;

  // todo
  handleTextareaChange: IEvent;
  handleRadioChange: IEvent;
  rateData: string;
  textData: string;
}
