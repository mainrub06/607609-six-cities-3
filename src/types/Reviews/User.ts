export interface IUser {
  id: number;
  isPro: boolean;
  name: string;
  avatar: string;
}
