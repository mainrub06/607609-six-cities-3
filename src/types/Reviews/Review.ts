import { IUser } from "./User";

export interface IReview {
  id: number;
  rate: number;
  comment: string;
  date: string;
  user: IUser;
}
