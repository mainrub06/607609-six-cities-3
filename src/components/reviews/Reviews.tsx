import React from "react";
import ReviewsList from "../ReviewList/ReviewList";
import ReviewsForm from "../ReviewsForm/ReviewsForm";
import withFormData from "../../hocs/with-form-data/with-form-data";
import { AUTHORIZATION_STATUS } from "../../const";
import { IReview } from "../../types/Reviews/Review";
import { IReviewForm, IReviewFormSubmit } from "../../types/Reviews/ReviewForm";

export interface IFormWrapper {
  activeHotelId: string;
  handleSubmitFeedback(
    firstArg: IReviewFormSubmit,
    secondArg: string | undefined
  ): void;
  reviewsResponse: number;
}

interface IReviews extends IFormWrapper {
  reviews: Array<IReview>;
  authStatus: string;
}

const ReviewsFormWrapper = withFormData<IFormWrapper>(ReviewsForm);

const Reviews = ({
  reviews,
  authStatus,
  handleSubmitFeedback,
  activeHotelId,
  reviewsResponse,
}: IReviews) => {
  return (
    <>
      {reviews && (
        <section className="property__reviews reviews">
          <h2 className="reviews__title">
            Reviews &middot;
            {reviews.length > 0 && (
              <span className="reviews__amount">{reviews.length}</span>
            )}
          </h2>
          <ReviewsList reviews={reviews} />
          {authStatus === AUTHORIZATION_STATUS.AUTH && (
            <ReviewsFormWrapper
              reviewsResponse={reviewsResponse}
              activeHotelId={activeHotelId}
              handleSubmitFeedback={handleSubmitFeedback}
            />
          )}
        </section>
      )}
    </>
  );
};

export default Reviews;
