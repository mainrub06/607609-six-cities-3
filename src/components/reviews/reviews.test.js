import React from "react";
import renderer from "react-test-renderer";
import Reviews from "./reviews.jsx";

const activeHotelId = `1`;

const REVIEWS = [
  {
    id: 1,
    rate: 2,
    comment: `review.comment`,
    date: `review.date`,
    user: {
      id: 1,
      isPro: false,
      name: `review.user.name`,
      avatar: `review.user.avatar_url`
    }
  },
  {
    id: 2,
    rate: 3,
    comment: `review.comment`,
    date: `review.date`,
    user: {
      id: 1,
      isPro: false,
      name: `review.user.name`,
      avatar: `review.user.avatar_url`
    }
  }
];

it(`Render snapshot <Reviews/>`, () => {
  const tree = renderer
    .create(
        <Reviews activeHotelId = {activeHotelId} review = {REVIEWS[0]}/>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
