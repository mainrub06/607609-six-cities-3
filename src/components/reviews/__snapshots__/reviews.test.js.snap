// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render snapshot <Reviews/> 1`] = `null`;
